static/js/typeahead_helper.js

```diff
@@ -95,6 +95,29 @@ function prefix_sort(query, objs, get_item) {
 
 }
 
+function split_by_subscribers(people) {
+    var subscribers = [];
+    var non_subscribers = [];
+    var current_stream = compose.stream_name();
+
+    if (current_stream === "") {
+        // For the home page, everyone is a subscriber.
```

Comment:
For the PM issue, I could work on that as a separate commit for https://github.com/zulip/zulip/issues/2008